### PR TITLE
Prism error on file format mismatch

### PR DESCRIPTION
**Pull request by curlup**

### CODE REVIEW

File: cidc_schemas/prism.py [hunk 1]
```diff
@@ -387,12 +389,23 @@ def _process_field_value(
     return changes, files
 
 
+def _get_file_ext(fname):
+    return fname.rsplit(".")[-1]
```
Comment (jacoblurye) [inline cidc_schemas/prism.py]: Python provides functionality for extracting extensions in stdlib's `pathlib`:
```python
from pathlib import Path
Path("/foo/a.b.txt").suffix == ".txt"

File: cidc_schemas/prism.py [hunk 2]
```diff
@@ -387,12 +389,23 @@ def _process_field_value(
...
     local_path: str, uuid: str, field_def: dict, format_context: dict
 ):
 
     gs_key = field_def["gcs_uri_format"].format_map(format_context)
 
+    expected_extension = _get_file_ext(gs_key)
+    provided_extension = _get_file_ext(local_path)
+    if provided_extension != expected_extension:
+        raise ParsingException(
+            f"Expected {'.'+expected_extension} for {field_def['key_name']!r} but got {'.'+provided_extension!r} instead."
```
Comment (jacoblurye) [inline cidc_schemas/prism.py]: I'd provide the full provided file path in this error message so the user doesn't have to search around.

File: cidc_schemas/template.py [hunk 3]
```diff
@@ -255,7 +257,9 @@ def _add_coerce(field_def: dict) -> dict:
 
                     # populate lookup
                     # TODO .lower() ?
-                    key_lu[column_key] = _add_coerce(column_def)
+                    key_lu[column_key] = _add_coerce(
+                        dict(column_def, key_name=column_key)
+                    )
```
Comment (jacoblurye) [inline cidc_schemas/template.py]: Do these changes relate to this PR?

File: tests/test_prism.py [hunk 4]
```diff
@@ -1066,6 +1066,68 @@ def test_merge_stuff():
...
+    """ Tests prism alert on different extensions of a local file vs gcs_uri """
+
+    load_workbook = MagicMock(name="load_workbook")
+    monkeypatch.setattr("openpyxl.load_workbook", load_workbook)
+    workbook = load_workbook.return_value = MagicMock(name="workbook")
+    wb = {"files": MagicMock(name="files")}
+    workbook.__getitem__.side_effect = wb.__getitem__
+    workbook.sheetnames = wb.keys()
+    cell = namedtuple("cell", ["value"])
+    wb["files"].iter_rows.return_value = [
```
Comment (jacoblurye) [inline tests/test_prism.py]: IIRC, this is copied verbatim from another test. Might it be worth it to refactor this out into its own function (that, e.g., takes just the `iter_rows.return_value` as its argument and returns relevant mocks)?

File: cidc_schemas/prism.py (hunk 2 above)
Comment (curlup) [reply cidc_schemas/prism.py]: well, but then we were trying to not include filenames in our systems whatsoever due to possible HIPAA issues?

File: cidc_schemas/prism.py (hunk 1 above)
Comment (curlup) [reply cidc_schemas/prism.py]: I don't like creating an object for now reason then just calling a split function.

File: cidc_schemas/template.py (hunk 3 above)
Comment (curlup) [reply cidc_schemas/template.py]: Yes, it's used for error formatting later

**Pull request merged**